yoyodyne/models/base.py

```diff
@@ -256,7 +257,8 @@ def predict_step(
         Returns:
             torch.Tensor: indices of the argmax at each timestep.
         """
-        predictions = self(batch)
+        batch.target.padded = None
```

Comment:
I notice that `batch` and `predictions` aren't passed in. Is it supposed to work like that, getting them from context? Seems scary.

Reply:
Also wouldn't you want to null out the whole target portion of the batch and not just the padded vector?

Reply:
> I notice that batch and predictions aren't passed in. Is it supposed to work like that, getting them from context? Seems scary.

I don't follow what you mean here.

> Also wouldn't you want to null out the whole target portion of the batch and not just the padded vector?

Ideally, I think yes. I can't remember which model behavior forced me to do this. Will retrace my steps and update (either with a fix, or clearer justification :D)

Reply:
Not sure if this is what you mean here, but thinking about it more, it might be a bad idea to force remove targets here? On the one hand, I don't think `predict` should ever let the model see the targets. On the other hand, this should probably be handled elswhere that is less implicit.

The problem is that currently there is not obvious place in the code to say "don't use the target" (I guess setting teacher forcing to False handles this...). And I think its reasonable to pass in a test file with targets and expect the code to not use them at runtime when predicting.

Reply:
I mean i don't understand why this doesn't complain that `batch` is undefined. The following snippet of Python should give a `NameError` on `baz` because `baz` hasn't been defined. Similarly, `batch` and `prediction` are undefined in this method. We don't rely on this kind of scoping anywhere else in the entire project to my knowledge: we pass in the arguments we refer to.

```python
def foo(bar):
   return baz + baz
```

Yes, we should pass this as a kwarg and not forcibly remove the target.

Reply:
Ah that's right, sorry to waste your time.